Src/DependencyCollector/Shared/Implementation/HttpDiagnosticSourceListener.cs

```diff
@@ -100,6 +105,81 @@ protected virtual void Dispose(bool disposing)
                 this.disposed = true;
             }
         }
+
```

Comment:
With the PropertyFetcher, reflection is slow only for for the first time you fetcvh property from the same kind on anonymous object.
It creates a delegate that fetches the property and caches it. The next fetching from the same kind of the object is fast.

DiagnosticSource sends anonymous objects and reflection is the only way to get property value.

Before this change is accidentally worked because the code that created anonymous object lived in the same assembly and was accessible. So `dynamic` worked.
It's not the case now when HttpHandlerDiagnosticListener is gone to DiagnosticSource